saleor/graphql/discount/mutations/sale/sale_delete.py

```diff
@@ -43,22 +48,28 @@ class Meta:
     def perform_mutation(  # type: ignore[override]
         cls, root, info: ResolveInfo, /, *, id: str
     ):
-        instance = cls.get_node_or_error(info, id, only_type=Sale)
-        previous_catalogue = fetch_catalogue_info(instance)
-        manager = get_plugin_manager_promise(info.context).get()
+        object_id = cls.get_global_id_or_error(id, "Sale")
+        promotion = models.Promotion.objects.get(old_sale_id=object_id)
```

Comment:
I'm thinking about raising some specific error for running this mutation with the promotion id (instead of `old_sale_id`) something like: `this is promotion, use "PromotionDelete" mutation`.